src/natcap/invest/hra.py

```diff
@@ -1,3253 +1,2424 @@
+                            subregion_block, subregion_nodata)) &
+                        habitat_mask])
+
+                for subregion_id in local_unique_subregions:
+                    subregion_mask = (subregion_block == subregion_id)
+                    try:
+                        stats = subregion_stats[subregion_id]
+                    except KeyError:
+                        # If subregion_id is not in the subregion_stats dict,
+                        # we need to initialize it.
+                        stats = {}
+                        for prefix in ('E', 'C', 'R'):
+                            for suffix, initial_value in [
+                                    ('MIN', float('inf')), ('MAX', 0),
+                                    ('SUM', 0), ('N_PIXELS', 0)]:
+                                stats[f'{prefix}_{suffix}'] = initial_value
+
+                            for r_stat in ('R_N_HIGH', 'R_N_MEDIUM', 'R_N_LOW',
+                                           'R_N_ANY'):
+                                stats[r_stat] = 0
+
+                    for prefix, raster in [('E', e_raster),
+                                           ('C', c_raster),
+                                           ('R', r_raster)]:
+                        valid_pixels = _read_block(raster, block_info,
+                                                   subregion_mask)
+                        if valid_pixels.size == 0:
+                            # If no valid pixels, fall back to defaults.
+                            pixel_min = float('inf')
+                            pixel_max = pixel_sum = n_pixels = 0
+                        else:
+                            pixel_min = numpy.min(valid_pixels)
+                            pixel_max = numpy.max(valid_pixels)
+                            pixel_sum = numpy.sum(valid_pixels)
+                            n_pixels = valid_pixels.size
+
+                        if pixel_min < stats[f'{prefix}_MIN']:
+                            stats[f'{prefix}_MIN'] = pixel_min
+
+                        if pixel_max > stats[f'{prefix}_MAX']:
+                            stats[f'{prefix}_MAX'] = pixel_max
+
+                        stats[f'{prefix}_SUM'] += pixel_sum
+                        stats[f'{prefix}_N_PIXELS'] += n_pixels
+
+                    # the percentage of high/medium/low classification is a
+                    # per-category count.
+                    classes_block = _read_block(classes_raster, block_info,
+                                                subregion_mask)
+                    stats['R_N_HIGH'] += numpy.count_nonzero(
+                        classes_block == 3)
+                    stats['R_N_MEDIUM'] += numpy.count_nonzero(
+                        classes_block == 2)
+
+                    # Old HRA included 0-value pixels in  the R_%_LOW bucket so
+                    # we do here as well.
+                    stats['R_N_LOW'] += numpy.count_nonzero(classes_block <= 1)
```

Comment:
Yeah, this looks contradictory but it should work as expected.  We're calling `numpy.count_nonzero` on a boolean index, where the boolean index is 1 where `classes_block <= 1`.  So therefore pixels in `classes_block` with a pixel value of 0 would be tallied as `True` and therefore `1`.

To avoid this confusion, I've updated this to use `numpy.sum` in 38839d1c9